package/MDAnalysis/analysis/msd.py

```diff
@@ -383,10 +388,12 @@ def _single_frame(self):
         ]
 
     def _conclude(self):
-        if self.fft:
+        if self.fft == True:
             self._conclude_fft()
-        else:
+        elif self.non_linear != True:
             self._conclude_simple()
+        elif self.non_linear:
+            self._conclude_non_linear()
```

Comment:
Let's group this logically:
```
if LINEAR
   do either FFT or normal
else
   do non-linear
```

This keeps the logic MUCH clearer and is more robust – you don't have an `else` so with weird settings, conclude will not do anything.